Taskqueue user execute (#3685)
mendix/docs: content/refguide/task-queue.md

#### 2.1.3 Context in Task Queues

> For microflows and Java actions which are running in a task queue, the circumstances in which the task runs changes slightly in the following ways:

Reviewer: Is there a reason that you don't want to use the word `context`?
For me, context is more accurate, has a technical meaning that customers and internal developers understand, but also has a general meaning in English that non-technical people can also understand.
Circumstances feels very abstract and carries, for me, the implication that it is something accidental rather than deliberately designed.

Author: Yes, that was a deliberate change on my part, because I felt context was used ambiguously here:
1. context can refer to the Mendix session context object
2. context can refer to the circumstances that change between invoking in the foreground and the background
The subsequent bullet items are not about changes to the context object. In the other places in the text context is accurate, however, because there it is referring to the Mendix session context.

Reviewer: Thanks for the explanation. I hadn't thought about the second point. You mean, for example, that an entity might have been deleted or modified in the meantime?

I wonder if `conditions` might be a better word? But I'll leave it as `circumstances` for now.

Author: Yes, that would be good as well. I'll leave that up to you.

Author: Yes, that would be good as well. I've updated the text.

Author: And with the second point I was referring to the changes as elaborated in the bullet items.

  .executeInBackground(context, "AModule.SomeQueueName");
```

> The method `executeInBackground` takes two parameters: a context and a queue name. The context is only used for creating the task; executing the task will be done with a new, but equivalent context. See the [API documentation](https://apidocs.rnd.mendix.com/9/runtime/com/mendix/core/Core.html#microflowCall(java.lang.String)) for more information.

Reviewer: Do you think we could expand this "new, but equivalent context"? Perhaps to say:
> this means that the task will have the same role(s) as the context passed and so will be subject to the same security.

> The method `executeInBackground` takes two parameters: a context and a queue name. The context is only used for creating the task; executing the task will be done with a new, but equivalent context. See the [API documentation](https://apidocs.rnd.mendix.com/9/runtime/com/mendix/core/Core.html#microflowCall(java.lang.String)) for more information.

Reviewer: I think we should link to the explanation of context here:
```suggestion
The method `executeInBackground` takes two parameters: a context and a queue name. The context is only used for creating the task; executing the task will be done with a [new, but equivalent context](#context). See the [API documentation](https://apidocs.rnd.mendix.com/9/runtime/com/mendix/core/Core.html#microflowCall(java.lang.String)) for more information.
```

  .executeInBackground(context, "AModule.SomeQueueName");
```

> The method `executeInBackground` takes two parameters: a context and a queue name. The context is only used for creating the task; executing the task will be done with a new, but equivalent context. See the [API documentation](https://apidocs.rnd.mendix.com/9/runtime/com/mendix/core/Core.html#userActionCall(java.lang.String)) for more information.

Reviewer: and here:
```suggestion
The method `executeInBackground` takes two parameters: a context and a queue name. The context is only used for creating the task; executing the task will be done with a [new, but equivalent context](#context). See the [API documentation](https://apidocs.rnd.mendix.com/9/runtime/com/mendix/core/Core.html#userActionCall(java.lang.String)) for more information.
```

`System.ProcessedQueueTasks` objects are never deleted. The user is free to delete them when desired.

> ### 2.7 Execution Context

Reviewer: Add an anchor here for the links above.
```suggestion
### 2.7 Execution Context{#context}
```

Prior to Mendix 9.6 tasks were always executed in a *sudo* context, even if the scheduling microflow had **Apply entity access** set to *true* (see [Microflow Properties](microflow) for more information). As of Mendix 9.6 this behavior has been deprecated and tasks now run in an equivalent context to the one in which they were scheduled. This has the following effect:

* When a user is logged in, the task will be executed in a new context for the same named user. This context will be the same as if the user is logged in.
* When no user is logged in, the task will be executed in a new anonymous context. This context will be for a new anonymous user with the same language and timezone as the original user.
* When a system session is used to schedule the task (using the Java API), the task will be executed in a new system context.

> Projects containing task queues that were created before Mendix 9.6 will get a deprecation warning on the system context. This warning can be remedied in the project settings for the runtime. By choosing *no* for *System context tasks* the deprecation warning goes away and the tasks will executed in an equivalent context to the one they were created in.

Reviewer: i.e. here? 
![image](https://user-images.githubusercontent.com/35492184/132824413-73bdf7a0-1efc-4458-b08e-209f011e70f9.png)

Author: Yes, it will look like this: 
![image](https://user-images.githubusercontent.com/58431933/132825868-b2710370-7898-4c5b-94ad-11d2ca8e4277.png)

Reviewer: Thanks for the screenshots.
I don't think we need them in the document, but perhaps we can help the user to find the setting by expanding the text a bit?

```suggestion
Projects containing task queues that were created before Mendix 9.6 will get a deprecation warning in the log: `The project setting 'System context tasks' is deprecated`.
You can remove this warning in the **Runtime** tab of the app **Settings** in Studio Pro. Set **System context tasks** to *no* to execute tasks in an equivalent context to the one they were created in and remove the warning.
```

{{% alert type="warning" %}}

> After choosing *no*, you cannot switch back to *yes* because executing tasks in system contexts (unless scheduled from a system session) is deprecated.

Reviewer: ```suggestion
You will be asked to confirm this change as, after choosing *no*, you cannot switch back to *yes* because executing tasks in system contexts (unless scheduled from a system session) is deprecated.
```
